pkg/server/commands/list_objects.go

```diff
@@ -287,6 +288,13 @@ func (q *ListObjectsQuery) evaluate(
 	return nil
 }
 
+func sendObject(object string, foundCount *uint32, maxResults uint32, resultsChan chan<- ListObjectsResult) {
+	if foundCount != nil && atomic.AddUint32(foundCount, 1) > maxResults {
+		return
```

Comment:
Curious that none of our unit test/integration test hits this condition.

Reply:
Yeah... i'm missing a test 🤔

Reply:
Ok should be covered now.